### ISSUE TITLE
Problem with conditional, multiple useSWRInfinite on version 0.5.x

### ISSUE THREAD

**Issue by tomaszczura** [vercel/swr#1124]
# Bug report

## Description / Observed Behavior

# Bug report

## Description / Observed Behavior
I have code, which uses three `useSWRInfinite` to load data from three endpoints. Loading each next data type depends on the previous typoe response. It has some kind of flag, which indicates, if the next data type should be loaded:

```
 const {
    data: items1Data,
  } = useSWRInfinite(
    getItems1key(pageId),
    fetchItems1()
  );

  const {
    data: items2Data,
  } = useSWRInfinite(
    items1Data.loadItems2 ? getItems2key(pageId) : () => null,
    fetchItems2()
  );

  const {
    data: items2Data,
  } = useSWRInfinite(
    items2Data.loadItems3 ? getItems3key(pageId) : () => null,
    fetchItems3()
  );
```
This worked well for me when I in version 0.4.x. But when I switched to version 0.5.0 and any newer, I observed a very strange behavior:
The code still works when I load a page first time. But then, when I switch to another page with different `pageId` (so I load items for different object, but still use the same container component), all data which has conditional cache key (here - `items2Data` and `items3Data`) have value `1`. Example:

I go to a page `A`, and load data. Cache key are: `items1A`, `items2A`, `items3A`, and all data is correctly fetched (they all should be arrays of objects). Then I click somehwere on UI to switch to page B, where cache keys are: `items1B`, `null`, `null`. In that case `items2Data` and `items3Data` (for which I got `null` keys) have value `1`. It's not `undefined`, not `null` - always `1`, if previous key was not null.  

Any idea why it has so strange value? And why it works for versions below 0.5.x?

**Comment by shuding** [comment]
This issue should have been fixed in #1096, which was released in the beta version. Please give it a try by upgrading with `yarn add swr@beta`!

**Comment by shuding** [comment]
If you are still seeing an issue, please consider creating a reproduction with this template: https://codesandbox.io/s/jolly-breeze-igcq9 (or a similar demo) that I can look into.
